### PR TITLE
New module: onyx_aaa_authentication

### PR DESCRIPTION
**Pull request by Sara-Touqan**
##### SUMMARY

Supporting AAA Authentication configs in Ansible.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

onyx_aaa_authentication
##### ADDITIONAL INFORMATION

```paste below
This module supports the config params for aaa authentication.
```

### CODE REVIEW

File: lib/ansible/modules/network/onyx/onyx_aaa_authentication.py [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+---
+module: onyx_aaa_authentication
+version_added: "2.10"
+author: "Sara Touqan (@sarato)"
+short_description: Configures AAA authentication parameters
+description:
+  - This module provides declarative management of AAA authentication protocol params
+    on Mellanox ONYX network devices.
+options:
+  auth_login_method1:
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_aaa_authentication.py:23]: auth_login_methods should be an array (length 1 to 4)

File: lib/ansible/modules/network/onyx/onyx_aaa_authentication.py [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+from ansible.module_utils.network.onyx.onyx import BaseOnyxModule
+
+
+class OnyxAaaAuthenticationModule(BaseOnyxModule):
+
+    def init_module(self):
+        """ initialize module
+        """
+        element_spec = dict(
+            auth_login_method1=dict(type='str', choices=['local', 'ldap', 'radius', 'tacacs+']),
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_aaa_authentication.py:180]: define login methods constants

File: lib/ansible/modules/network/onyx/onyx_aaa_authentication.py [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+            auth_login_method2=dict(type='str', choices=['local', 'ldap', 'radius', 'tacacs+']),
+            auth_login_method3=dict(type='str', choices=['local', 'ldap', 'radius', 'tacacs+']),
+            auth_login_method4=dict(type='str', choices=['local', 'ldap', 'radius', 'tacacs+']),
+            auth_fail_delay_time=dict(type='int'),
+            auth_track_enabled=dict(type='bool'),
+            track_downcase_enabled=dict(type='bool'),
+            reset_for=dict(type='str', choices=['all', 'user']),
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_aaa_authentication.py:187]: add relations between the reset parameter.
user_name is mandatory when reset_for is user

File: lib/ansible/modules/network/onyx/onyx_aaa_authentication.py [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+            unlock_time=dict(type='int'),
+            max_failure_account=dict(type='int')
+        )
+        argument_spec = dict()
+        argument_spec.update(element_spec)
+        self._module = AnsibleModule(
+            argument_spec=argument_spec,
+            supports_check_mode=True)
+
+    def validate_auth_params(self):
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_aaa_authentication.py:204]: this validation can be done using spec relationships

File: lib/ansible/modules/network/onyx/onyx_aaa_authentication.py [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+    def _set_aaa_running_config(self, all_aaa_config):
+        curr_config_arr = []
+        aaa_config = all_aaa_config.get('Lines')
+        if aaa_config is None:
+            self._current_config['curr_config_arr'] = curr_config_arr
+            return
+        for runn_config in aaa_config:
+            curr_config_arr.append(runn_config.strip())
+
+        if 'aaa authentication attempts track downcase' in curr_config_arr:
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_aaa_authentication.py:243]: no need for another loop
you can check this in the previous loop

File: lib/ansible/modules/network/onyx/onyx_aaa_authentication.py [hunk 6]
```diff
@@ -0,0 +1,456 @@
...
+        curr_lockout = self._current_config.get("lockout_enabled")
+        lock_time = self._required_config.get("lock_time")
+        curr_lock_time = self._current_config.get("lock_time")
+        unlock_time = self._required_config.get("unlock_time")
+        curr_unlock_time = self._current_config.get("unlock_time")
+        max_failure_account = self._required_config.get("max_failure_account")
+        curr_max_failure = self._current_config.get("max_failure_account")
+
+        if curr_config_arr is not None:
+            if auth_login_method4 is not None:
```
Comment (samerd) [inline lib/ansible/modules/network/onyx/onyx_aaa_authentication.py:340]: see above suggestion to use array of auth methods

**Review by samerd — CHANGES_REQUESTED** [review]
thanks @Sara-Touqan 
please see my review

**Comment by Akasurde** [comment]
Hi @Sara-Touqan, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/mellanox.onyx<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**